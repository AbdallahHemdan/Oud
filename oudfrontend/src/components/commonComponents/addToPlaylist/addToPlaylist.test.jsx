import React from "react";
import AddToPlaylist from "./addToPlaylist";
import Enzyme, { shallow } from "enzyme";
import EnzymeAdapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";

Enzyme.configure({ adapter: new EnzymeAdapter() });
const props = {
  close: jest.fn(),
};

const setup = (props) => {
  return shallow(<AddToPlaylist {...props} />);
};

const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-testid="${val}"]`);
};
/*
Rendering tests
*/

describe("CreatePlaylist Component", () => {
  describe("testing that CreatePlaylist renders Correctly", () => {
    let component;
    beforeEach(() => {
      component = setup();
    });
    it("renders createPlaylist component", () => {
      component.setState({ createPlaylist: true });
      const wrapper = findByTestAttr(component, "createPlaylist");
      expect(wrapper.length).toBe(1);
    });
    it("renders createPlaylist component", () => {
      component.setState({ createPlaylist: false });
      const wrapper = findByTestAttr(component, "createPlaylist");
      expect(wrapper.length).toBe(0);
    });
    it("renders createPlaylist component", () => {
      const wrapper = findByTestAttr(component, "closeButton");
      expect(wrapper.length).toBe(1);
    });
    it("renders createPlaylist component", () => {
      const wrapper = findByTestAttr(component, "addToPlaylist");
      expect(wrapper.length).toBe(1);
    });
    it("renders createPlaylist component", () => {
      const wrapper = findByTestAttr(component, "first-wrapper");
      expect(wrapper.length).toBe(1);
    });
    it("renders createPlaylist component", () => {
      const wrapper = findByTestAttr(component, "createNew");
      expect(wrapper.text()).toBe("NEW PLAYLIST");
      expect(wrapper.length).toBe(1);
    });
    it("renders createPlaylist component", () => {
      const wrapper = findByTestAttr(component, "title");
      expect(wrapper.length).toBe(1);
      expect(wrapper.text()).toBe("Add To playlist");
    });
    it("renders createPlaylist component", () => {
      const wrapper = findByTestAttr(component, "second-wrapper");
      expect(wrapper.length).toBe(1);
    });
    it("renders createPlaylist component", () => {
      const wrapper = findByTestAttr(component, "cards-wrapper");
      expect(wrapper.length).toBe(1);
    });
    it("renders createPlaylist component", () => {
      const wrapper = findByTestAttr(component, "cards");
      expect(wrapper.length).toBe(0);
    });
  });
  describe("testing that CreatePlaylist renders Correctly", () => {
    let component;
    beforeEach(() => {
      component = setup(props);
      component.setState({ display: true });
    });
    it("close Button", () => {
      component.setState({ display: true });
      const wrapper = findByTestAttr(component, "closeButton");
      expect(component.state().display).toBeTruthy();
      wrapper.simulate("click");
      expect(component.state().display).toBeFalsy();
    });
    it("new playlist Button", () => {
      component.setState({ createPlaylist: false });
      const wrapper = findByTestAttr(component, "createNew");
      expect(component.state().createPlaylist).toBeFalsy();
      wrapper.simulate("click");
      expect(component.state().createPlaylist).toBeTruthy();
    });
  });
  describe("componentWillReceiveProps()", () => {
    it("change the display state", () => {
      const component = shallow(<AddToPlaylist display={true} />);
      expect(component.state().display).toBe(true);
      component.setProps({ display: false });
      expect(component.state().display).toBe(false);
    });
    it("does not change the display state", () => {
      const component = shallow(<AddToPlaylist display={true} />);
      expect(component.state().display).toBe(true);
      component.setProps({ display: true });
      expect(component.state().display).toBe(true);
    });
  });
  describe("snapshot test", () => {
    it("renders correctly", () => {
      const tree = renderer.create(<AddToPlaylist />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
