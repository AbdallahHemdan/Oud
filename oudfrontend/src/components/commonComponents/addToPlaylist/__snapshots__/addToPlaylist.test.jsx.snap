// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePlaylist Component snapshot test renders correctly 1`] = `
<div>
  <div
    className="createPlaylist hide"
    data-testid="addToPlaylist"
  >
    <button
      className="closeButton"
      data-testid="closeButton"
      onClick={[Function]}
    >
      <svg
        height="32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          close
        </title>
        <path
          d="M31.098 29.794L16.955 15.65 31.097 1.51 29.683.093 15.54 14.237 1.4.094-.016 1.508 14.126 15.65-.016 29.795l1.414 1.414L15.54 17.065l14.144 14.143"
          fill="#fff"
          fillRule="evenodd"
        />
      </svg>
    </button>
    <h1
      data-testid="title"
      id="createPlaylistTitle"
    >
      Add To playlist
    </h1>
    <button
      className="playButton"
      data-testid="createNew"
      onClick={[Function]}
    >
      NEW PLAYLIST
    </button>
    <div
      id="createPlaylistBigField"
    >
      <div
        className="wrapper"
        data-testid="first-wrapper"
        id="wrapperId"
      >
        <div
          className="wrapper_section_2"
          data-testid="second-wrapper"
          id="second-wrapperId"
        >
          <div
            className="cards"
            data-testid="cards-wrapper"
            id="cardsId"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
